skbio/diversity/_driver.py

```diff
@@ -180,9 +182,114 @@ def alpha_diversity(metric, counts, ids=None, validate=True, **kwargs):
     return pd.Series(results, index=ids)
 
 
+@deprecated(as_of='0.4.2-dev', until='0.5.0',
+            reason=('The return type is unstable. Developer caution is '
+                    'advised. The resulting DistanceMatrix object may include '
+                    'zeros when distance has not been calculated, and '
+                    'therefore can be misleading.'))
+def partial_beta_diversity(metric, counts, ids=None, validate=None,
```

Comment:
make `id_pairs` a positional argument. if a user wants all pairs they should be using `beta_diversity` since this function is unstable/deprecated.

so the new function sig is:

```python
def partial_beta_diversity(metric, counts, ids, id_pairs, validate=True, **kwargs):
    ...
```